### DISCUSSION TITLE
❓ Is `FLASHBOTS_SIGNATURE_PK` Required?

### DISCUSSION THREAD

**Discussion by MidnightLightning** [pcaversaccio/white-hat-frontrunning#2] [Q&A] [answered]
To submit a Flashbots bundle, it needs to be signed with a "unique ECDSA-secp256k1 key pair for Flashbots to identify you", which does gain reputation over time. However it doesn't need to be funded as a blockchain address, just exist as a signing key.

I think an enhancement to this script would be if the environment variable `FLASHBOTS_SIGNATURE_PK` is not set, instead of erroring out, it instead uses something like [`cast wallet new`](https://book.getfoundry.sh/reference/cli/cast/wallet/new) to make a new bundle-signing key, and saves out the private key as an updated config file, for subsequent runs to use?

**Comment by pcaversaccio** [comment, marked as answer]
I believe it's essential for everyone to understand the role of this private key and its intended use, so I prefer _not_ to make it optional. However, I've added a note in the `README` (via https://github.com/pcaversaccio/white-hat-frontrunning/pull/3) to clarify its role.

![image](https://github.com/user-attachments/assets/8bcb97ad-8311-4e27-807e-68504a34733c)
